### PR TITLE
pcs_cluster: add transport_options arg

### PR DESCRIPTION
**Pull request by atemp**
support pcs transport options

### CODE REVIEW

**Comment by OndrejHome** [comment]
Hi @atemp, thank you for using the pcs-modules-2 and the PR here.
I will have a look at it in my evening today and give you feedback.
Please let me know on which system(s) you have tested the code -I mainly interested if this works well in both `pcs-0.9` and `pcs-0.10` which can have some options positioned differently.

**Comment by atemp** [comment]
Your are right, i just checked it with CentOS 7, there is no such option. I added a check so it will throw an error if used with 0.9.

Btw. thanks for creating this module!

File: library/pcs_cluster.py [hunk 1]
```diff
@@ -49,6 +49,10 @@
     required: false
     default: default
     choices: ['default', 'udp', 'udpu', 'knet']
+  transport_options:
+    description:
+      - additional options for transports (must not be used with pcs 0.9)
```
Comment (OndrejHome) [inline library/pcs_cluster.py]: (low priority) I would rephrase it to something like "available only with `pcs-0.10`"

Comment (OndrejHome) [reply library/pcs_cluster.py]: (medium priority) "this option can be used only when `transport` options is specified (non-default)"

File: library/pcs_cluster.py [hunk 2]
```diff
@@ -91,6 +95,16 @@
     state: 'present'
   run_once: True
 
+- name: Create cluster with redundant corosync links and transport options
```
Comment (OndrejHome) [inline library/pcs_cluster.py]: (low priority) "- name: Create cluster with redundant corosync links and transport and link options"

File: library/pcs_cluster.py [hunk 3]
```diff
@@ -127,6 +141,7 @@ def run_module():
             cluster_name=dict(required=False),
             token=dict(required=False, type='int'),
             transport=dict(required=False, default="default", choices=['default', 'udp', 'udpu', 'knet']),
+            transport_options=dict(required=False, type='str'),
```
Comment (OndrejHome) [inline library/pcs_cluster.py]: (medium priority) add `default=''` so we don't need to use the 'if' condition later, default (if not specified) will be empty string

File: library/pcs_cluster.py [hunk 4]
```diff
@@ -182,15 +197,21 @@ def run_module():
     # if there is no cluster configuration and cluster should be created do 'pcs cluster setup'
     if state == 'present' and not (cluster_conf_exists or corosync_conf_exists or cib_xml_exists):
         result['changed'] = True
-        # create cluster from node list that was provided to module
+        # create cluster from node list that was provided to module        # create cluster from node list that was provided to module
```
Comment (OndrejHome) [inline library/pcs_cluster.py]: (low priority) this looks like duplicated comment, would be nice to remove it

File: library/pcs_cluster.py [hunk 5]
```diff
@@ -182,15 +197,21 @@ def run_module():
...
         if pcs_version == '0.9':
+            # if no transport_options are specified used empty string
+            if (module.params['transport_options']):
+                module.fail_json(msg="using transport_options is not supported with pcs 0.9")
             module.params['token_param'] = '' if (not module.params['token']) else '--token %(token)s' % module.params
             module.params['transport_param'] = '' if (module.params['transport'] == 'default') else '--transport %(transport)s' % module.params
             cmd = 'pcs cluster setup --name %(cluster_name)s %(node_list)s %(token_param)s %(transport_param)s' % module.params
         elif pcs_version == '0.10':
+            # if no transport_options are specified used empty string
```
Comment (OndrejHome) [inline library/pcs_cluster.py]: this can be handled by `default=''` for `transport_options` parameter as mentioned earlier which I think looks cleaner

**Comment by OndrejHome** [comment]
Hi @atemp, I just wanted to check if you would have time in upcoming week to have a look at my earlier feedback to this PR or if you would need more time (in which case please just let me know roughly your time plan here). 

Thank you

**Comment by atemp** [comment]
Sorry for the delay and thanks for your detailed review. I still playing around and are on it, i will try to work all your requests into my branch in the next few days.

**Comment by atemp** [comment]
Hi, i tried to address all of your points. I managed today to do a few test rollouts on CentOS 7.7 and 8.1 and here it looks okay.

**Comment by OndrejHome** [comment]
Hi @atemp, thank you for additional changes. All points looks to be addressed. I'm going to merge this PR now. I will be adding some more changes from `devel` later today or tomorrow and create a new tag for whole release. I will leave you message here once the tag is created.

Thank you for your contribution to these modules!

**Comment by OndrejHome** [comment]
Hi @atemp, this PR is now included in tag `22.0.0` and was also pushed to Ansible Galaxy.
